doc/api/module.md

```diff
@@ -1048,6 +1048,92 @@ returned object contains the following keys:
+replace inline type annotations with whitespace, and no type checking is
+performed. TypeScript features that depend on settings within `tsconfig.json`,
+such as paths or converting newer JavaScript syntax to older standards, are
+intentionally unsupported.
+
+To get fuller TypeScript support, including support for enums and namespaces
+and paths, see <https://nodejs.org/en/learn/getting-started/nodejs-with-typescript>.
+The built-in TypeScript support is designed to be lightweight and as fast as
+possible. By intentionally not supporting syntaxes that require JavaScript code
+generation, and by replacing stripped types with whitespace, Node.js can run
+TypeScript code without the need for source maps and with less overhead.
+
+### Determining module system
+
+Node.js supports both [CommonJS][] and [ES Modules][] syntax in TypeScript
+files. Node.js will not convert from one module system to another; if you want
+your code to run as an ES module, you must use `import` and `export` syntax,
+and if you want your code to run as CommonJS you must use `require` and
+`module.exports`.
+
+* `.ts` files will have their module system determined
+  [the same way as `.js` files.][] To use `import` and `export` syntax, add
+  `"type": "module"` to the nearest parent `package.json`.
+* `.mts` files will always be run as ES modules, similar to `.mjs` files.
+* `.cts` files will always be run as CommonJS modules, similar to `.cjs` files.
+* `.tsx` files are unsupported.
+
+As in JavaScript files, [file extensions are mandatory][] in `import` statements
+and `import()` expressions: `import './file.ts'`, not `import './file'`.
+Because of backward compatibility, file extensions are also mandatory in
+`require()` calls: `require('./file.ts')`, not `require('./file')`, similar to
+how the `.cjs` extension is mandatory in `require` calls in CommonJS files.
+
+The `tsconfig.json` option `allowImportingTsExtensions` will allow the
+TypeScript compiler `tsc` to type-check files with `import` specifiers that
+include the `.ts` extension.
+
+### Unsupported TypeScript features
+
+Since Node.js is only removing inline types, any TypeScript features that
+involve _replacing_ TypeScript syntax with new JavaScript syntax will error.
+This is by design. To run TypeScript with such features, see
+<https://nodejs.org/en/learn/getting-started/nodejs-with-typescript#running-typescript-code-in-nodejs>
+
+The most prominent unsupported features that require transformation are:
+
+* `Enum`
+* `experimentalDecorators`
+* `namespaces`
+
+In addition, Node.js does not read `tsconfig.json` files and does not support
+features that depend on settings within `tsconfig.json`, such as paths or
+converting newer JavaScript syntax into older standards.
+
+### Type stripping in node\_modules
+
+To avoid encouraging package authors to publish TypeScript only modules,
+Node.js will not support running TypeScript files in `node_modules`.
```

Comment:
It's not that we don't support it, if the user creates a loader that resolves `.ts` files inside of `node_modules/` as e.g. `typescript-module`, we should support it.
```suggestion
Node.js will by default refuse to handle TypeScript files inside `node_modules` directories.
```